### PR TITLE
allow user to set tools upload path

### PR DESCRIPTION
**Pull request by SwampDragons**
Addresses part of issue #9971 by allowing users to provide a local copy of the vmware tools.

### CODE REVIEW

**Review by sylviamoss — APPROVED** [review]
LGTM!! 👍

File: builder/vmware/common/tools_config.go [hunk 1]
```diff
@@ -18,12 +20,22 @@ type ToolsConfig struct {
 	// the upload path is set to `{{.Flavor}}.iso`. This setting is not used
 	// when `remote_type` is `esx5`.
 	ToolsUploadPath string `mapstructure:"tools_upload_path" required:"false"`
+	// The path on you local machine to fetch the vmware tools from. If this
+	// is not set but the tools_upload_flavor is set, then Packer will try to
+	// downlaod the vmware tools for your version of VMware from the internet.
```
Comment (azr) [inline builder/vmware/common/tools_config.go:25]: should we say that it is cached too ?

**Review by azr — APPROVED** [review]
LGTM ! Just had a tiny question 🙂

File: builder/vmware/common/tools_config.go (hunk 1 above)
Comment (nywilken) [inline builder/vmware/common/tools_config.go:25]:
```suggestion
	// download the vmware tools for your version of VMware from the internet.
```

**Review by nywilken — APPROVED** [review]
One small typo in the generated docs, but good to go. Nice stuff!

File: builder/vmware/common/tools_config.go (hunk 1 above)
Comment (SwampDragons) [reply builder/vmware/common/tools_config.go:25]: is it?

**Pull request merged**